pkg/controller/quota/quota.go

```diff
@@ -0,0 +1,187 @@
+		return err
+	}
+
+	// Create the quota request object and give calculate the standard numeric values
+	name, namespace, app := appInstance.Name, appInstance.Namespace, appInstance.Status.AppSpec
+	quotaRequest := &adminv1.QuotaRequestInstance{
+		ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace},
+		Spec: adminv1.QuotaRequestInstanceSpec{
+			Resources: adminv1.Resources{
+				Containers: len(app.Containers),
+				Jobs:       len(app.Jobs),
+				Volumes:    len(app.Volumes),
+				Images:     len(app.Images),
+			},
+		},
+	}
+
+	status := condition.Setter(appInstance, resp, apiv1.AppInstanceConditionQuotaAllocated)
+
+	// Add the more complex values to the quota request
+	addCompute(app.Containers, appInstance, quotaRequest)
+	addCompute(app.Jobs, appInstance, quotaRequest)
+	if err := addStorage(appInstance, quotaRequest); err != nil {
+		status.Error(err)
+		return err
+	}
+
+	resp.Objects(quotaRequest)
+	return nil
+}
+
+// addCompute adds the compute resources of the containers passed to the quota request.
+func addCompute(containers map[string]apiv1.Container, appInstance *apiv1.AppInstance, quotaRequest *adminv1.QuotaRequestInstance) {
+	// For each workload, add their memory/cpu requests to the quota request
+	for name, container := range containers {
+		var requirements corev1.ResourceRequirements
+		if specific, ok := appInstance.Status.Scheduling[name]; ok {
+			requirements = specific.Requirements
+		} else if all, ok := appInstance.Status.Scheduling[""]; ok {
+			requirements = all.Requirements
+		}
+
+		quotaRequest.Spec.Resources.CPU.Add(requirements.Requests["cpu"])
+		quotaRequest.Spec.Resources.Memory.Add(requirements.Requests["memory"])
+
+		// Recurse over any sidecars. Since sidecars can't have sidecars, this is safe.
+		addCompute(container.Sidecars, appInstance, quotaRequest)
+	}
+}
+
+// addStorage adds the storage resources of the volumes passed to the quota request.
+func addStorage(appInstance *apiv1.AppInstance, quotaRequest *adminv1.QuotaRequestInstance) error {
+	app := appInstance.Status.AppSpec
+
+	// Add the volume storage needed to the quota request. We only parse net new volumes, not
+	// existing ones that are then bound client-side.
+	for name, volume := range app.Volumes {
+		size := volume.Size
+		if bound, boundSize := boundVolumeSize(name, appInstance.Spec.Volumes); bound {
+			size = boundSize
+		}
+
+		// No need to proceed if the size is empty
+		if size == "" || size == "0" {
+			continue
+		}
+
+		parsedSize, err := resource.ParseQuantity(string(size))
+		if err != nil {
+			return err
+		}
+		quotaRequest.Spec.Resources.VolumeStorage.Add(parsedSize)
+	}
+
+	// Add the secrets needed to the quota request. We only parse net new secrets, not
+	// existing ones that are then bound client-side.
+	for name := range app.Secrets {
+		if boundSecret(name, appInstance.Spec.Secrets) {
+			continue
+		}
+		quotaRequest.Spec.Resources.Secrets = +1
```

Comment:
Most importantly, should this be `+=` instead of `= +`?
```suggestion
		if !boundSecret(name, appInstance.Spec.Secrets) {
			quotaRequest.Spec.Resources.Secrets += 1
		}
```

Reply:
That's super strange, I didn't even know that was supported behavior If I'm honest. Yeah I'll fix this, thank you for catching it.